signac/core/validators.py

```diff
@@ -0,0 +1,95 @@
+# Copyright (c) 2020 The Regents of the University of Michigan
+# All rights reserved.
+# This software is licensed under the BSD 3-Clause License.
+"""Validators for SyncedCollection API."""
+import warnings
+from collections.abc import Mapping
+from collections.abc import Sequence
+
+from ..errors import InvalidKeyError
+from ..errors import KeyTypeError
+from .synced_collection import NUMPY
+
+if NUMPY:
+    import numpy
```

Comment:
I would rather copy the try/except snippet into this file, rather than import `synced_collection`.
```suggestion

try:
    import numpy
    NUMPY = True
except ImportError:
    NUMPY = False
```